leetcode/src/124.c

```diff
@@ -0,0 +1,35 @@
+/**
+ * Definition for a binary tree node.
+ * struct TreeNode {
+ *     int val;
+ *     struct TreeNode *left;
+ *     struct TreeNode *right;
+ * };
+ */
+
+#define max(a,b) (((a)>(b))?(a):(b))
+
+int recursiveSolve(struct TreeNode* node, int* result){
+    if (node == NULL){
+        return 0;
+    }
+
+    int leftSum = max(recursiveSolve(node->left, result), 0);
+    int rightSum = max(recursiveSolve(node->right, result), 0);
+
+    // Check if it's possible to make a maximum path from left right and current node
+    int maxValueNode = node->val + leftSum + rightSum;
+    *result = max(maxValueNode, *result);
+
+    // Choose the max sum val path 
+    return node->val + max(leftSum, rightSum);
+}
+
+// Depth First Search
+// Runtime: O(n), n - the number of nodes in tree.
+// Space: O(1)
+int maxPathSum(struct TreeNode* root){
+    int result = -2147483648;

```

Comment:
Haha - this is fine but generally for (magic) constants like these, it is really encouraged to name them 😄

In this case, this is acting like a lower bound, so potentially defining this as

```suggestion
    const int LOWER_BOUND = -2147483648
    int result = LOWER_BOUND;
```

Or something along those lines 😄